docs/source/deployment/amazon_emr_serverless.md

```diff
@@ -0,0 +1,244 @@
+Here is the `entrypoint.py` entrypoint script:
+
+```python
+import sys
+from <PACKAGE_NAME>.__main__ import main
+
+main(sys.argv[1:])
+
+```
+
+Replace `<PACKAGE_NAME>` with your package name.
+
+### Resources
+For more details, refer to the following resources:
+
+- [Package a Kedro project](https://docs.kedro.org/en/stable/tutorial/package_a_project.html#package-a-kedro-project)
+- [Run a packaged project](https://docs.kedro.org/en/stable/tutorial/package_a_project.html#run-a-packaged-project)
+- [Customizing an EMR Serverless image](https://docs.aws.amazon.com/emr/latest/EMR-Serverless-UserGuide/application-custom-image.html)
+- [Using custom images with EMR Serverless](https://docs.aws.amazon.com/emr/latest/EMR-Serverless-UserGuide/using-custom-images.html)
+
+## Setup
+
+### Prerequisites
+You must create an S3 bucket to store data for EMR Serverless.
+
+### Infrastructure
+
+1. **Create a private repository in ECR**. See details on how to do so [in the AWS documentation](https://docs.aws.amazon.com/AmazonECR/latest/userguide/repository-create.html). You can use the default settings. This repository will be used to upload the EMR Serverless custom image.
+
+2. **Create an EMR Studio**.
+
+- Go to the AWS Console > EMR > EMR Serverless.
+- Make sure you are in the correct region where you want to create your resource.
+- Then, click "Get started" > "Create and launch EMR Studio".
+
+3. **Create an application**.
+
+- Type: Spark
+- Release version: `emr-6.10.0`
+- Architecture: `x86_64`
+- Application setup options > "Choose custom settings"
+- Custom image settings > "Use the custom image with this application"
+- For simplicity, use the same custom image for both Spark drivers and executors
+- Provide the ECR image URI; it must be located in the same region as the EMR Studio
+
+You may customise other settings as required, or even change the release version and architecture options
+(along with making the changes in the custom image).
+
+```{warning}
+This code base has only been tested with the above options.
+```
+
+See the AWS EMR Serverless documentation for [more details on creating an application](https://docs.aws.amazon.com/emr/latest/EMR-Serverless-UserGuide/studio.html).
+
+### IAM
+
+1. **Create a job runtime role**. Follow the instructions under ["Create a job runtime role"](https://docs.aws.amazon.com/emr/latest/EMR-Serverless-UserGuide/getting-started.html). This role will be used when submitting a job to EMR Serverless. The permissions defined in the policy also grant access
+to an S3 bucket (specify the S3 bucket you have created).
+
+2. **Grant access to ECR repository**.
+
+- Go to ECR in the AWS console.
+- Click into the repository you created earlier, then on the left under Repositories > Permissions.
+- Click "Edit policy JSON" and paste [this policy](https://docs.aws.amazon.com/emr/latest/EMR-Serverless-UserGuide/application-custom-image.html) under "Allow EMR Serverless to access the custom image repository".
```

Comment:
👍🏼 This is great docs.

Reply:
Haha thanks! I actually struggled with this myself, couldn't find where/how to change the permission 🙈

Reply:
> I actually struggled with this myself
The best kind of docs :) Other people are likely get stumbled.

https://www.writethedocs.org/guide/writing/beginners-guide-to-docs/#you-will-be-using-your-code-in-6-months